### PR TITLE
Add information modal to JSON fields

### PR DESCRIPTION
**Pull request by Unraveler**
Adds an information modal to the JSON Fields, containing a small documentation about the field.
Given examples can be coppied to clipboard.

Please review @terrestris/devs 

https://user-images.githubusercontent.com/9415123/208658645-b88b6471-c90d-4ab7-9284-5a7869b92c21.mp4

### CODE REVIEW

**Review by dnlkoch — CHANGES_REQUESTED** [review]
Generally this looks very promising and very useful for users - thanks so far @Unraveler!

There are many `TODO`s and references to a project (in the example JSONs) left so it looks like a WIP?

File: src/Component/InformationModal/information/layer/FeaturesInformation.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          ]
+        },
+        "properties": {
+          "prop0": "value0",
+          "prop1": { "this": "that" }
+        }
+      }
+    ]
+  }
+`;
```
Comment (dnlkoch) [inline src/Component/InformationModal/information/layer/FeaturesInformation.tsx:46]: Having the examples included in the code makes it quite hard to override it in projects. Maybe it's easier to include it in external files, e.g. the existing formconfig files? Defaults may be derived from the OpenAPI schema?

File: src/i18n/translations.ts [hunk 2]
```diff
@@ -1,6 +1,29 @@
...
+            sourceConfig: 'Datenquelle',
+            features: 'Features'
+          },
+          user: {
+            details: 'Details',
+            clientConfig: 'Keycloak Details',
+            providerDetails: 'Datenquelle'
+          }
+        }
+      },
```
Comment (dnlkoch) [inline src/i18n/translations.ts:26]: Can we reuse the locales from the formconfig files instead?

File: src/Component/InformationModal/information/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      }
+      return finalSchema;
+    });
+  }
+  return [{
+    propertyName: 'TODO',
+    description: 'TODO',
+    example: 'TODO',
+    dataType: 'TODO',
+    required: 'TODO',
```
Comment (dnlkoch) [inline src/Component/InformationModal/information/helpers.ts:54]: Some TODOs are left here.

File: src/Component/InformationModal/InformationModal.tsx [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+      title: 'Data type',
+      dataIndex: 'dataType',
+      key: 'dataType' + idCode
+    },
+    {
+      title: 'Required',
+      dataIndex: 'required',
+      key: 'required' + idCode
+    },
+  ];
```
Comment (dnlkoch) [inline src/Component/InformationModal/InformationModal.tsx:91]: The `title` values might be added to the `i18n` file.

File: src/Component/InformationModal/InformationModal.tsx [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+      case 'user-providerDetails':
+        return t('InformationModal.title.user.providerDetails');
+
+      default:
+        return [{
+          propertyName: 'TODO',
+          description: 'TODO',
+          example: 'TODO',
+          dataType: 'TODO',
+          required: 'TODO',
```
Comment (dnlkoch) [inline src/Component/InformationModal/InformationModal.tsx:124]: There are some `TODO`s left.

Comment (Unraveler) [reply src/Component/InformationModal/InformationModal.tsx:124]: These are not TODOs, actually. They are just fallback values. But I'll return `undefined` instead in order to skip showing the table as a whole.

File: src/Component/InformationModal/information/helpers.ts (hunk 3 above)
Comment (Unraveler) [reply src/Component/InformationModal/information/helpers.ts:54]: These are not TODOs, actually. They are just fallback values. But I'll return `undefined` instead in order to skip showing the table as a whole.

File: src/Component/InformationModal/information/layer/FeaturesInformation.tsx (hunk 1 above)
Comment (Unraveler) [reply src/Component/InformationModal/information/layer/FeaturesInformation.tsx]: The examples have been removed. They are now coming through the API (see https://github.com/terrestris/shogun/pull/624 ). It also possible to pass a custom example through the formconfigs.

**Pull request closed without merging**